guides/contexts.md

```diff
@@ -8,1060 +8,1313 @@
 ```
 
-We built a query for fetching the current page given its ID which we pass to `Repo.update_all`. Ecto's `Repo.update_all` allows us to perform batch updates against the database, and is perfect for atomically updating values, such as incrementing our views count. The result of the repo operation returns the number of updated records, along with the selected schema values specified by the `select` option. When we receive the new page views, we use `put_in(page.views, views)` to place the new view count within the page.
+To show our completed order, we displayed the order's user, followed by the line item listing with product title, quantity, and the price we "transacted" when completing the order, along with the total price.
 
-With our context function in place, let's make use of it in our CMS page controller. Update your `show` action in `lib/hello_web/controllers/cms/page_controller.ex` to call our new function:
+Our last addition will be to add the "complete order" button to our cart page to allow completing an order. Add the following button to the botto of your cart show template in `lib/hello_web/templates/cart/show.html.eex`:
 
+```eex
+  <b>Total</b>: <%= currency_to_str(ShoppingCart.total_cart_price(@cart)) %>
 
-```elixir
-def show(conn, %{"id" => id}) do
-  page =
-    id
-    |> CMS.get_page!()
-    |> CMS.inc_page_views()
-
-  render(conn, "show.html", page: page)
-end
++ <%= link "complete order", to: Routes.order_path(@conn, :create), method: :post %>
+<% end %>
 ```
 
-We modified our `show` action to pipe our fetched page into `CMS.inc_page_views/1`, which will return the updated page. Then we rendered our template just as before. Let's try it out. Refresh one of your pages a few times and watch the view count increase.
-
-We can also see our atomic update in action in the ecto debug logs:
+We added a link with `method: :post` to send a POST request to our `OrderController.create` action. If we head back to our cart page at `http://localhost:4000/cart` and complete an order, we'll by greeted by our rendered template:
 
 ```
-[debug] QUERY OK source="pages" db=3.1ms
-UPDATE "pages" AS p0 SET "views" = p0."views" + $1 WHERE (p0."id" = $2)
-RETURNING p0."views" [1, 3]
+Thank you for your order!
+
+User uuid: 08964c7c-908c-4a55-bcd3-9811ad8b0b9d
+Metaprogramming Elixir (2) - $15.00
+Total price: $30.00
 ```
 
-Good work!
+Nice work! We haven't added payments, but we can already see our how `ShoppingCart` and `Orders` context splitting is driving us towards maintainable solution. With our cart items separate from our order line items, we well equipped in the future to add payment transactions, cart price detection, and more.
 
-As we've seen, designing with contexts gives you a solid foundation to grow your application from. Using discrete, well-defined APIs that expose the intent of your system allows you to write more maintainable applications with reusable code.
+Great work!
 
 ## FAQ
 
```

Comment:
This sections needs to either be updated or removed, as it references to the implementation of the now replaced CMS.